distribution/src/assembly/binary-tarball.xml

```diff
@@ -0,0 +1,133 @@
+  <includeBaseDirectory>true</includeBaseDirectory>
+
+  <moduleSets>
+
+    <!-- Module java-driver-core and its dependencies -->
+    <moduleSet>
+      <useAllReactorProjects>true</useAllReactorProjects>
+      <includes>
+        <include>com.datastax.oss:java-driver-core</include>
+      </includes>
+      <binaries>
+        <outputDirectory>lib/core</outputDirectory>
+        <unpack>false</unpack>
+        <dependencySets>
+          <dependencySet>
+            <outputDirectory>lib/core</outputDirectory>
+            <excludes>
+              <!--
+                For some reason, we need to exclude all other modules here, even though our
+                moduleSet targets core only
+              -->
+              <exclude>com.datastax.oss:java-driver-query-builder</exclude>
+            </excludes>
+            <useTransitiveFiltering>true</useTransitiveFiltering>
+          </dependencySet>
+        </dependencySets>
+      </binaries>
+    </moduleSet>
+
+    <!-- Module java-driver-query-builder and its dependencies -->
+    <moduleSet>
+      <useAllReactorProjects>true</useAllReactorProjects>
+      <includes>
+        <include>com.datastax.oss:java-driver-query-builder</include>
+      </includes>
+      <binaries>
+        <outputDirectory>lib/query-builder</outputDirectory>
+        <unpack>false</unpack>
+        <dependencySets>
+          <dependencySet>
+            <excludes>
+              <exclude>com.datastax.oss:java-driver-core</exclude>
+              <!-- Don't reinclude dependencies that core also has -->
+              <exclude>com.datastax.oss:java-driver-shaded-guava</exclude>
+              <exclude>com.github.stephenc.jcip:jcip-annotations</exclude>
+              <!-- TODO exclude modules other than core when they get added -->
+            </excludes>
+            <useTransitiveFiltering>true</useTransitiveFiltering>
+          </dependencySet>
+        </dependencySets>
+      </binaries>
+    </moduleSet>
+
+    <!--sources for all modules-->
+    <moduleSet>
+      <useAllReactorProjects>true</useAllReactorProjects>
+      <includes>
+        <include>com.datastax.oss:java-driver-core</include>
+        <include>com.datastax.oss:java-driver-query-builder</include>
+      </includes>
+      <binaries>
+        <unpack>false</unpack>
+        <attachmentClassifier>sources</attachmentClassifier>
+        <outputFileNameMapping>${module.artifactId}-${module.version}-src.zip
+        </outputFileNameMapping>
+        <outputDirectory>src</outputDirectory>
+        <excludes>
+          <exclude>*</exclude>
+        </excludes>
+      </binaries>
+    </moduleSet>
+
+  </moduleSets>
+
+  <fileSets>
+
+    <fileSet>
+      <directory>target/apidocs</directory>
+      <outputDirectory>apidocs</outputDirectory>
+    </fileSet>
+
+    <fileSet>
+      <directory>..</directory>
+      <outputDirectory>.</outputDirectory>
+      <includes>
+        <include>README*</include>
+        <include>LICENSE*</include>
```

Comment:
I noticed that there was no `LICENSE` file in the tar, but then I realized we don't have a `LICENSE` file currently present for 4.x, we should add that.